modules/core/shared/src/main/scala/lucuma/core/validation/InputValidWedge.scala

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.validation
+
+import cats.data.NonEmptyChain
+import cats.syntax.all._
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.Positive
+import eu.timepit.refined.types.numeric.PosBigDecimal
+import eu.timepit.refined.types.string.NonEmptyString
+import lucuma.core.optics._
+import monocle.Iso
+import monocle.Prism
+
+/**
+ * Convenience version of `ValidWedge` when the error type is `NonEmptyChain[NonEmptyString]` and
+ * `T` is `String`.
+ */
+object InputValidWedge {
+
+  /**
+   * Build an `InputValidWedge` that's always valid and doesn't normalize or format
+   */
+  val id: InputValidWedge[String] = ValidWedge.id
+
+  /**
+   * Build an `InputValidWedge` from `getValid` and `reverseGet` functions
+   */
+  def apply[A](
+    getValid:   String => EitherErrors[A],
+    reverseGet: A => String
+  ): InputValidWedge[A] =
+    ValidWedge(getValid, reverseGet)
+
+  /**
+   * Build an `InputValidWedge` from a `Format`
+   */
+  def fromFormat[A](
+    format:       Format[String, A],
+    errorMessage: NonEmptyString = "Invalid format"
+  ): InputValidWedge[A] =
+    ValidWedge(
+      format.getOption.andThen(_.toRight(errorMessage).toEitherErrors),
+      format.reverseGet
+    )
+
+  /**
+   * Build an `InputValidWedge` from a `Prism`
+   */
+  def fromPrism[A](
+    prism:        Prism[String, A],
+    errorMessage: NonEmptyString = "Invalid value"
+  ): InputValidWedge[A] =
+    fromFormat(Format.fromPrism(prism), errorMessage)
+
+  /**
+   * Build an `InputValidWedge` from an `Iso`
+   */
+  def fromIso[A](iso: Iso[String, A]): InputValidWedge[A] =
+    ValidWedge(
+      (iso.get _).andThen(_.asRight),
+      iso.reverseGet
+    )
+
+  /**
+   * Build a `InputValidWedge` for `BigDecimal` truncated to `Dec` decimals.
+   */
+  def truncatedBigDecimal(decimals: DigitCount): InputValidWedge[BigDecimal] =
+    InputValidWedge(
+      InputValidSplitEpi.bigDecimal.getValid
+        .andThen(_.map(_.setScale(decimals.value, scala.math.BigDecimal.RoundingMode.HALF_UP))),
+      bd =>
+        s"%.${decimals.value}f"
+          .format(bd)
+          .replaceAll("^-0\\.(0+)$", "0.$1") // Remove negative 0
+    )
+
+  // We can't build a generic truncatedRefinedBigDecimal. Formatting may cause the refinement to break.
+  // For example: a truncated PosBigDecimal may turn a small value into a 0, which is no longer positive.
```

Comment:
Do we still have a way to deal with things like `HourAngle`, which has a `BigDecimal` refined to the interval `-5.0` to `5.0`. In the old version we used a `SplitEpi` called `unsafeRefinedBigDecimal` to edit it to 1 decimal place. It was unsafe because, as you say here, it could not be used for the general case, but it would work if you constrained the decimals proper.

Reply:
I made specific instances for those cases in explore, which are also tested now. Maybe it makes sense to move them here too so they can be used by the API?

Reply:
Cool. I guess if the API is going to use these validators, they'll need to validate the elevation range stuff. I don't know what the API is doing, though.

Reply:
https://github.com/gemini-hlsw/lucuma-tmp-api/blob/master/modules/core/src/main/scala/lucuma/odb/api/model/ElevationRangeModel.scala#L88

I'm probably not going to update the `tmp-api` much from here but it may be of use in the real ODB.